### PR TITLE
Add Overlay Toggle to Wintertodt Plugin

### PR DESCRIPTION
**Pull request by JonnoUK**
Requested feature
https://github.com/runelite/runelite/issues/11482

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtConfig.java [hunk 1]
```diff
@@ -128,4 +128,15 @@ default boolean notifyBrazierOut()
 	{
 		return true;
 	}
+
+	@ConfigItem(
+			position = 9,
+			keyName = "Overlay",
+			name = "Client Overlay Toggle",
+			description = "Toggle the overlay for this plugin"
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtConfig.java:136]: These lines are all indented one level too far.
```suggestion
		position = 9,
		keyName = "Overlay",
		name = "Client Overlay Toggle",
		description = "Toggle the overlay for this plugin"
```

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtOverlay.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ private WintertodtOverlay(WintertodtPlugin plugin)
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		if (!plugin.isInWintertodt())
+		if (!plugin.isInWintertodt() || plugin.clientOverlayToggle == false)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtOverlay.java:56]: `boolean == false` is the same as `!boolean`, use the latter form.

Also, don't store the config state in plugin, just inject the config and directly call the config method here instead.

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtConfig.java (hunk 1 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtConfig.java:136]: Also, the names given to the added config fields/method are not good. I would recommend naming the method and keyName `displayOverlay`, the name `Display Overlay`, and the description `Display an overlay which shows your activity status and number of points and kindling held in your inventory.`

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtOverlay.java [hunk 3]
```diff
@@ -50,10 +50,13 @@ private WintertodtOverlay(WintertodtPlugin plugin)
 		getMenuEntries().add(new OverlayMenuEntry(RUNELITE_OVERLAY_CONFIG, OPTION_CONFIGURE, "Wintertodt overlay"));
 	}
 
+	@Inject
+	private WintertodtConfig config;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtOverlay.java:54]: Inject this in the constructor in the same way that `plugin` is injected in this class.

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtPlugin.java [hunk 4]
```diff
@@ -126,7 +126,7 @@
 	private Instant lastActionTime;
 
 	private int previousTimerValue;
-
+
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtPlugin.java:129]: Remove this change
```suggestion

```

**Comment by Adam-** [comment]
added in eae4e0b6a06618aec9f9df5d4be8bdadddd26f53

**Pull request closed without merging**